### PR TITLE
Import Fixes

### PR DESCRIPTION
**Pull request by sravfeyn**
@nickpell 
Fixes reports-fixture missing from restore and adds a test.

There are some unknowns (see comments) but the issue gets resolved after these import arrangements

### CODE REVIEW

File: custom/bihar/reports/indicators/indicators.py [hunk 1]
```diff
@@ -275,6 +274,7 @@ def get_indicator_set(self, slug):
 class IndicatorSet(object):
 
     def __init__(self, spec):
+        from django.utils.datastructures import SortedDict
```
Comment (sravfeyn) [inline custom/bihar/reports/indicators/indicators.py]: This fix is not really related to the bug, but I am pretty sure, bihar fixture items are missing due to an import error here. So this fixes that issue (I am not sure if bihar is active or not)

I do not understand why this is an issue though, it just got fixed with this.

File: corehq/ex-submodules/casexml/apps/phone/fixtures.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def __init__(self):
         if hasattr(settings, "FIXTURE_GENERATORS"):
             for group, func_paths in settings.FIXTURE_GENERATORS.items():
                 self._generator_providers[group] = filter(None, [
-                    to_function(func_path) for func_path in func_paths
+                    to_function(func_path, failhard=True) for func_path in func_paths
```
Comment (sravfeyn) [inline corehq/ex-submodules/casexml/apps/phone/fixtures.py]: Because `failhard` was set to False, `to_function('corehq.apps.app_manager.fixtures.report_fixture_generator')` was returning None and so this was missing from the restore. Same issue happens for `"custom.bihar.reports.indicators.fixtures.generator"`

**Review by nickpell — APPROVED** [review]
Nice, thanks for fixing!

File: corehq/apps/reports_core/filters.py [hunk 3]
```diff
@@ -184,6 +181,7 @@ def year_param_name(self):
 
     @property
     def years(self):
+        from corehq.apps.userreports.reports.filters.values import SHOW_ALL_CHOICE
```
Comment (sravfeyn) [inline corehq/apps/reports_core/filters.py]: These are bunch of circular import issues, as `userreports` module uses functions defined in `reports_core` module. This is the only place where `reports_core` uses functions defined in `userreports`

**Comment by sravfeyn** [comment]
I am interested to dig a bit more to understand what's going on, I will come back to this latter though (I spent lot of time on this)

File: corehq/ex-submodules/casexml/apps/phone/fixtures.py (hunk 2 above)
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/phone/fixtures.py]: great call

**Review by benrudolph — APPROVED** [review]
